contracts/facets/FundFacet.sol

```diff
@@ -201,22 +201,41 @@ contract FundFacet is IFund {
     function payContribution(uint termId) external {
         LibFundStorage.Fund storage fund = LibFundStorage._fundStorage().funds[termId];
 
-        // Get the beneficiary for this cycle
-        address currentBeneficiary = IGetters(address(this)).getCurrentBeneficiary(termId);
-
         require(
-            fund.currentState == LibFundStorage.FundStates.AcceptingContributions,
+            fund.currentState == LibFundStorage.FundStates.AcceptingContributions ||
+                fund.currentState == LibFundStorage.FundStates.CycleOngoing,
             "Wrong state"
         );
         require(fund.isParticipant[msg.sender], "Not a participant");
-        require(currentBeneficiary != msg.sender, "Beneficiary doesn't pay");
-        require(!fund.paidThisCycle[msg.sender], "Already paid for cycle");
-        require(
-            !fund.isExemptedOnCycle[fund.currentCycle].exempted[msg.sender],
-            "Participant is exempted this cycle"
-        );
 
-        _payContribution(termId, msg.sender, msg.sender);
+        bool payNextCycle;
+
+        if (fund.currentState == LibFundStorage.FundStates.AcceptingContributions) {
+            // Get the beneficiary for this cycle
+            address currentBeneficiary = IGetters(address(this)).getCurrentBeneficiary(termId);
+
+            require(currentBeneficiary != msg.sender, "Beneficiary doesn't pay");
+            require(!fund.paidThisCycle[msg.sender], "Already paid for cycle");
+            require(
+                !fund.isExemptedOnCycle[fund.currentCycle].exempted[msg.sender],
+                "Participant is exempted this cycle"
+            );
+        } else {
```

Comment:
Please add a check here that prevents the user for paying "next" cycle when the current cycle is the final cycle. If we don't do this we could risk funds getting stuck.

Reply:
Not necessary because on the last term when `closeFundingPeriod` is called the state changes to `FundClosed` instead of `CycleOngoing` so it reverts with `Wrong state` but add new tests to check that on commit e600a39